Adding a guide for contributors (#1433)
wix/react-native-ui-lib: CONTRIBUTING.md

# Contributing

> If you got here it means you want to dedicate some of your time and help us with making this library better.  

Reviewer: ```suggestion
If you got here it means you want to dedicate some of your time and help us improve this library.  
```

> So first, thank you ❤️

Reviewer: ```suggestion
So first of all, thank you ❤️
```

When contributing to this repository, please first discuss the change you wish to make via [issue](https://github.com/wix/react-native-ui-lib/issues/new/choose),
[Discord channel](https://discord.gg/2eW4g6Z), or any other method with the owners of this repository before making a change. 

> ## Pull Request Process

Author: Also worth mentioning PR prefixes (fix, feat, etc..)

1. First make sure the environment is working and synced with master. For installation details go [here](https://github.com/wix/react-native-ui-lib/blob/master/markdowns/getting-started/setup.md#demo-app)
2. Before submitting a PR we suggest running `npm run prepush` command that verifies our lint, TS and tests were not broken.
3. Please don't change our PR template.

> - 3.1 In Description add everything that can help the reviewer and help the reviewing process. 

Reviewer: ```suggestion
- 3.1 In the Description section add everything that can help the reviewer and the reviewing process, like a description of the issue and the way you decided to go about it, screenshots or gifs, etc.
```

> - 3.2 In the Changelog it's important a message for the library users - this message will be included in our release notes

Reviewer: ```suggestion
- 3.2 The Changelog section is important. Write a message for the library users - this message will be included in our release notes.
```

4. We try to make the PR review process as quick as possible, please be patient

> 

Reviewer: ```suggestion
5. Please don't submit automatic formatting changes to the files as it results in cumbersome code reviewing.
```

Author: But we do want them to keep the code formatted according to our guidelines.. no?

Reviewer: Yes. but if they auto format it will take their environment's settings, not ours

Reviewer: I'm not sure that works... If the user has some other formatting tool configured in his environment, wouldn't it override the project's format configuration?

Author: Maybe, I think we'll learn more after a few PRs and see how it goes. 
And then we can add a more relevant message here about formatting..

Reviewer: Sound good

### Code Standards

- Readability and clean code above all - we believe a clean code is easier to maintain and refactor when needed. 
- Deciding on an API is not trivial but we always aim to keep it generic and intuitive - feel free to consult with us.
- Our documentation is derived from our components' prop comments, make sure to add a clear description when adding new props. 

> - When possible, consider adding tets for a new functionality you add.

Reviewer: ```suggestion
- When possible, consider adding tests for the new functionality you added.
```

> 

Reviewer 2: 1. Maybe worth adding controlled / uncontrolled preference.
2. It's possible always to have tests 🧪, at least unit (obviously not mandatory).
3. Clean code - means to separate the ui from the logic? worth adding?

Author: 1. done, added another section in Code Standards 
2. not sure I understand, is this a question? or a suggestion to rephrase the sentence?  (: 
3. Added a bullet for that, thank

Reviewer 2: Great :)
2. it's now more clear you referring to unit tests (`..tests for the new functionality you added`)

## Typescript Status
Currently most of our code is migrated to typescript though we still have some leftovers of javascript code and manual typings.  

> Till we complete the migration you require to do the following for new TS files

Reviewer: ```suggestion
Until we complete the migration you are required to do the following for new TS files
```
